modules: memfault-firmware-sdk: migrate coredump from PM to devicetree (#28820)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Memfault integration
--------------------

* Added devicetree support for the internal flash coredump backend (:kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP`).
  The backend now locates its storage from a fixed partition labelled ``memfault_coredump_partition`` when Partition Manager is disabled, while continuing to support Partition Manager when enabled.

> * Added the :ref:`snippet-memfault-coredump` snippet, which supplies a per-board partition overlay for flash-backed coredump storage on common Nordic DKs and Thingy:91 / Thingy:91 X.

Reviewer: ```suggestion
* Added:
  
  * Devicetree support for the internal flash coredump backend (:kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP`).
    The backend now locates its storage from a fixed partition labeled ``memfault_coredump_partition`` when Partition Manager is disabled, while continuing to support Partition Manager when enabled.
  * The :ref:`snippet-memfault-coredump` snippet, which supplies a per-board partition overlay for flash-backed coredump storage on common DKs and Thingy:91 or Thingy:91 X.
```

Memfault integration
--------------------

* Added:

  * Devicetree support for the internal flash coredump backend (:kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP`).
    The backend now locates its storage from a fixed partition labeled ``memfault_coredump_partition`` when Partition Manager is disabled, while continuing to support Partition Manager when enabled.

>   * The :ref:`snippet-memfault-coredump` snippet, which supplies a per-board partition overlay for flash-backed coredump storage on common DKs and Thingy:91 or Thingy:91 X.

Reviewer 2: ```suggestion
  * The :ref:`snippet-memfault-coredump` snippet that supplies a board-specific partition overlay for flash-backed coredump storage on common DKs and Thingy:91 or Thingy:91 X.
```
